app/models.py

```diff
@@ -1,182 +1,57 @@
+    notifications: Mapped[list["Notification"]] = relationship(back_populates="user")
 
+    @property
+    def name(self):
+        return f"{self.family_name} {self.given_name}"
 
-async def create_notification(notification: Notification, db_session: AsyncSession) -> Notification:
-    db_session.add(notification)
-    await db_session.commit()
-    await db_session.refresh(notification)
-    return notification
 
+class Registration(Base):
+    __tablename__ = "registration"  # type: ignore
 
-async def get_notification_by_id_and_user(
-    notification_id: int, user: User, db_session: AsyncSession
-) -> Notification:
-    query = select(Notification).where(
-        col(Notification.id) == notification_id, col(Notification.user) == user
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_id: Mapped[int] = mapped_column(ForeignKey("ami_user.id"))
+    user: Mapped[User] = relationship(back_populates="registrations")
+    subscription: Mapped[dict[str, Any]] = mapped_column(JsonB, nullable=True)
+    created_at: Mapped[datetime.datetime] = mapped_column(
+        DateTimeUTC(timezone=True),
+        default=lambda: datetime.datetime.now(datetime.timezone.utc),
     )
-    result = await db_session.exec(query)
-    try:
-        return result.one()
-    except NoResultFound as e:
-        raise NotFoundException(
-            detail=f"Notification with id {notification_id!r} not found for the user"
-        ) from e
 
 
-async def update_notification(db_session: AsyncSession, notification: Notification) -> Notification:
-    db_session.add(notification)
-    await db_session.commit()
-    await db_session.refresh(notification)
-    return notification
+class Notification(Base):
+    __tablename__ = "notification"  # type: ignore
 
-
-#### REGISTRATIONS
-
-
-async def get_registration_by_user_and_subscription(
-    subscription: dict[str, Any], db_session: AsyncSession, user: User
-) -> Registration | None:
-    query = select(Registration).where(
-        col(Registration.user) == user, col(Registration.subscription) == subscription
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_id: Mapped[int] = mapped_column(ForeignKey("ami_user.id"))
+    user: Mapped[User] = relationship(back_populates="notifications")
+    message: Mapped[str]
+    sender: Mapped[str | None]
+    title: Mapped[str | None]
+    date: Mapped[datetime.datetime] = mapped_column(
+        DateTimeUTC(timezone=True),
+        default=lambda: datetime.datetime.now(datetime.timezone.utc),
     )
-    result = await db_session.exec(query)
-    existing_registration = result.first()
-    return existing_registration
-
-
-async def create_registration(registration: Registration, db_session: AsyncSession) -> Registration:
-    db_session.add(registration)
-    await db_session.commit()
-    await db_session.refresh(registration)
-    return registration
+    unread: Mapped[bool] = mapped_column(default=True)

```

Comment:
Very small nit: would it make more sense, semantically, to have a `read` boolean? I don't know, if feels kind of backwards to have `unread`?